### ISSUE TITLE
testDeferredInsertBatching may have assertion failure on table write counter

### ISSUE THREAD

**Issue by tedyu** [yugabyte/yugabyte-db#6034]
Jira Link: [DB-5397](https://yugabyte.atlassian.net/browse/DB-5397)
Saw the following failure in one run:
```
m1|pid8997|:17255|http://127.248.125.27:29852 I1013 22:13:06.249850  9271 catalog_manager.cc:3730] Servicing IsDeleteTableDone request for table id 000030af000030008000000000004060: totally deleted
ts1|pid9023|:10527|http://127.146.53.87:32439 I1013 22:13:06.250481  9373 client-internal.cc:643] Deleted table 000030af000030008000000000004060
2020-10-13 22:13:06,280 (main) [ERROR - org.yb.BaseYBTest$1$2.logEventDetails(BaseYBTest.java:240)] YB Java test failed: class="org.yb.pgsql.TestPgForeignKeyBatching", method="testDeferredInsertBatching"
java.lang.AssertionError: expected:<21> but was:<20>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:645)
        at org.junit.Assert.assertEquals(Assert.java:631)
        at org.yb.AssertionWrappers.lambda$assertEquals$37(AssertionWrappers.java:580)
        at org.yb.AssertionWrappers.wrapAssertion(AssertionWrappers.java:45)
        at org.yb.AssertionWrappers.assertEquals(AssertionWrappers.java:580)
        at org.yb.pgsql.TestPgForeignKeyBatching.testDeferredInsertBatching(TestPgForeignKeyBatching.java:159)
```
As of commit 14842968989200008af74b79bb3103c6cad43af1

[DB-5397]: https://yugabyte.atlassian.net/browse/DB-5397?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by tedyu** [comment]
[org.yb.pgsql.TestPgForeignKeyBatching-output.txt](https://github.com/yugabyte/yugabyte-db/files/5374528/org.yb.pgsql.TestPgForeignKeyBatching-output.txt)

**Comment by d-uspenskiy** [comment]
Fixed in context of #17776
